yt_dlp/extractor/generic.py

```diff
@@ -3693,6 +3703,13 @@ def _real_extract(self, url):
         if tvopengr_urls:
             return self.playlist_from_matches(tvopengr_urls, video_id, video_title, ie=TVOpenGrEmbedIE.ie_key())
 
+        # Look for ert.gr webtv embeds
+        ertwebtv_urls = list(ERTWebtvEmbedIE._extract_urls(webpage))
+        if ertwebtv_urls:
+            return self.playlist_from_matches(
+                ertwebtv_urls, video_id, video_title, ie=ERTWebtvEmbedIE.ie_key(),
+                url_transparent=True, title=video_title)
```

Comment:
I just realized how weird this is. You are extracting all embeds from the page, and then setting all the videos' title to the page title? How does that make sense?

Reply:
The player page (matching the URL) does not provide any title whatsoever. A page will usually only embed a single video, yet this certainly can not be guaranteed.

Reply:
In that case, we can check the len of `ertwebtv_urls`. then if only one video is found, return url_transparent; and if multiple embeds, return playlist (without transparent)

Reply:
Sure. We have to call `playlist_from_matches`, or anyway something that calls `_proto_relative_url`, so as to qualify URLs. You mean something like this?

```python
        if ertwebtv_urls:
            ertwebtv_parent_info = {'title': video_title} if len(ertwebtv_urls) > 1 else {}
            return self.playlist_from_matches(
                ertwebtv_urls, video_id, video_title, ie=ERTWebtvEmbedIE.ie_key(),
                url_transparent=True, **ertwebtv_parent_info)
```

Reply:
Or perhaps it suffices to toggle `url_transparent` parameter value:

```python
        if ertwebtv_urls:
            return self.playlist_from_matches(
                ertwebtv_urls, video_id, video_title, ie=ERTWebtvEmbedIE.ie_key(),
                url_transparent=(len(ertwebtv_urls) > 1), title=video_title)
```

Reply:
No. like this:
```python
        if len(ertwebtv_urls) == 1:
            return self.url_result(self._proto_relative_url(ertwebtv_urls[0]), video_title=video_title, transparent=True)
        elif ertwebtv_urls:
            return self.playlist_from_matches(
                ertwebtv_urls, video_id, video_title, ie=ERTWebtvEmbedIE.ie_key())
```

I'm pretty sure the `_proto_relative_url` isn't even needed for url_result (The core code sanitizes the URL). But no harm in having it - perhaps it can be moved inside url_result too

Reply:
The issue is with the changes made to `playlist_from_matches`. Propogating kwargs into the video dict doesn't make sense in general and is counter to how `playlist_result` works

Reply:
I can agree the tweaks to `playlist_from_matches` and `url_result` as part of af08027 are not ideal, but anything else would seem to require wider, breaking changes.

Reply:
I don't quite understand why the utils tweak from master e.g. consumes positional parameters, but anyway this is now irrelevant to this PR.